packages/react-aria-components/test/Menu.test.tsx

```diff
@@ -1056,4 +1050,259 @@ describe('Menu', () => {
       expect(menu).not.toBeInTheDocument();
     });
   });
+
+  describe('portalContainer', () => {
+    function InfoMenu(props) {
+      return (
+        <UNSTABLE_PortalProvider getContainer={() => props.container.current}>
+          <MenuTrigger>
+            <Button aria-label="trigger" />
+            <Popover>
+              <Menu aria-label="Test">
+                <MenuItem id="1">One</MenuItem>
+                <MenuItem id="">Two</MenuItem>
+                <MenuItem id="3">Three</MenuItem>
+              </Menu>
+            </Popover>
+          </MenuTrigger>
+        </UNSTABLE_PortalProvider>
+      );
+    }
+
+    function App() {
+      let container = React.useRef(null);
+      return (
+        <>
+          <InfoMenu container={container} />
+          <div ref={container} data-testid="custom-container" />
+        </>
+      );
+    }
+
+    it('should render the menu in the portal container', async () => {
+      let {getByRole, getByTestId} = render(
+        <App />
+      );
+
+      let button = getByRole('button');
+      await user.click(button);
+
+      expect(getByRole('menu').closest('[data-testid="custom-container"]')).toBe(getByTestId('custom-container'));
+    });
+  });
+});
+
+// better to accept items from the test? or just have the test have a requirement that you render a certain-ish structure?
```

Comment:
I'm personally leaning towards the test having a requirement that you render a certain structure of menu, would make it easier to have others reuse the tests for their own implementation. If we aren't interested in that (it would be a lot of work to take on), then maybe we could accept items/button label/any other standardization from the test perhaps

Reply:
Yeah, I'd like to decouple more from the react testing library render or React in general, but right now it's still baked in pretty hard thanks to `act` most specifically.
Goal for another time, but hopefully this helps set us up for it eventually.